# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../blackbox/workspace-label.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

With a composition manager running, the toolbar does not get expose events on
workspace switch. Thus explicitly redraw the toolbar workspace label to
show the correct workspace number.

  - Rene Rebe <rene@exactcode.de>

--- blackbox-0.70.1/src/Screen.cc	2005-10-18 10:07:22.000000000 +0200
+++ blackbox-0.70.1-fixed/src/Screen.cc 2008-07-14 17:59:55.000000000 +0200
@@ -553,6 +553,9 @@
   XUnmapWindow(_blackbox->XDisplay(), empty_window);
 
   _blackbox->XUngrabServer();
+
+  if (_toolbar)
+    _toolbar->redrawWorkspaceLabel();
 }
 
 
